utilities/source_setup/gitall.py

```diff
@@ -19,66 +19,75 @@
 config = json.loads(open(os.path.join(os.path.dirname(__file__), 'config.json')).read())
 
 def incorrectInput(argv, msg):
-	print msg
-	sys.exit()
+    print(msg)
+    sys.exit()
 
 def main(argv):
 
-	command = []
-	if(len(argv) == 0):
-		incorrectInput(argv, 'Too few paramaters')
-
-	elif(argv[0] == 'push'):
-		command = ['git','push',argv[1],argv[2]]
-
-	elif(argv[0] == 'add'):
-		command = ['git','add',argv[1]]
-
-	elif(argv[0] == 'commit'):
-		command = ['git','commit',argv[1],argv[2]]
-
-	elif(argv[0] == 'branches'):
-		command = ['git','rev-parse','--abbrev-ref','HEAD']
-
-	elif(argv[0] == 'reset'):
-        	command = ['git','reset','--hard','HEAD']
-
-	elif(argv[0] == 'status'):
-		command = ['git',argv[0]]
-
-	elif(argv[0] == 'remote'):
-		for repo in config['repos']:
-			print repo['name']+'  '+subprocess.check_output(['git','remote','add','mlolson','https://github.com/mlolson/'+repo['name']+'.git'], cwd = os.path.join(config['sourcesdir'], repo['name']))
-		return
-
-	elif(argv[0] == 'checkout'):
-		if(len(argv) == 2):
-			command = ['git','checkout',argv[1]]
-		elif(len(argv) == 3):
-			command = ['git','checkout',argv[1],argv[2]]
-		else:
-			incorrectInput(argv, 'Expected <=3 paramaters')
-
-
-	elif(argv[0] == 'pull' or argv[0] == 'fetch'):
-		if(len(argv) == 1):
-			command = ['git',argv[0]]
-		elif(len(argv) ==2):
-			command = ['git',argv[0],argv[1]]
-		elif(len(argv) ==3):
-			command = ['git',argv[0],argv[1],argv[2]]
-		else:
-			incorrectInput(argv, 'Too many paramaters')
-
-	else:
-		incorrectInput(argv, 'Unrecognized command')
-
-	for repo in config['repos']:
-		try:
-			print repo['name']+'  '+subprocess.check_output(command, cwd = os.path.join(config['sourcesdir'], repo['name']))
-		except:
-			print "Error -- trying next repo"
-
+    command = []
+    if len(argv) == 0:
+        incorrectInput(argv, 'Too few paramaters')
+
+    elif argv[0] == 'push':
+        command = ['git', 'push', argv[1], argv[2]]
+
+    elif argv[0] == 'add':
+        command = ['git', 'add', argv[1]]
+
+    elif argv[0] == 'commit':
+        command = ['git' ,'commit' ,argv[1], argv[2]]
+
+    elif argv[0] == 'branches':
+        command = ['git', 'rev-parse', '--abbrev-ref', 'HEAD']
+
+    elif argv[0] == 'reset':
+        command = ['git','reset','--hard','HEAD']
+
+    elif argv[0] == 'status':
+        command = ['git', argv[0]]
+
+    elif argv[0] == 'remote':
+        for repo in config['repos']:
+            if repo['name'] == "geppetto-application":
+                cwd = os.path.join(config['sourcesdir'], 'org.geppetto.frontend', 'src', 'main', 'webapp')
+            else:
+                cwd = os.path.join(config['sourcesdir'], repo['name'])
+            message = subprocess.check_output(['git','remote','add','mlolson','https://github.com/mlolson/' + repo['name'] + '.git'], cwd=cwd)
```

Comment:
@rodriguez-facundo can mlolson become origin? also two other things 1) does the if repo['name'] == "geppetto-application" works also if the folder was called say `webapp`? 2) what if there is geppetto client inside the application, can we check and update that too?

Reply:
1. Idk why that mlolson was there. I did not want to touch it because I did not understand the utility of it. `origin` will already be there. Maybe if geppetto is a fork, you can replace `mlolson` by `upstream`.
2. geppetto-application is the name of the repo but the folder location is webapp.
3. If you checkout xx in geppetto-application, then package.json will be modified to point yo the right geppetto-client branch.